mono/mini/mini-exceptions.c

```diff
@@ -2413,6 +2413,34 @@ print_stack_frame_to_string (StackFrameInfo *frame, MonoContext *ctx, gpointer d
 
 #ifndef MONO_CROSS_COMPILE
 
+static void print_process_map ()
+{
+#ifdef __linux__
+	FILE *fp = fopen ("/proc/self/maps", "r");
+	char line [256];
+
+	if (fp == NULL) {
+		mono_runtime_printf_err ("no /proc/self/maps, not on linux?\n");
+		return;
+	}
+
+	mono_runtime_printf_err ("/proc/self/maps:");
+
+	while (fgets (line, sizeof (line), fp)) {
+		// strip newline
+		size_t len = strlen (line) - 1;
+		if (*line && line [len] == '\n')
```

Comment:
IMO `*line` doesn't convey the intent very well. I think it would be clearer if you had:
```
size_t len = strlen (line);
if (len > 0 && line [len - 1] == '\n')
  line [len - 1] = '\0';
```